src/main/java/hwr/oop/riddler/model/Sudoku.java

```diff
@@ -1,14 +1,67 @@
 package hwr.oop.riddler.model;
 
 import hwr.oop.riddler.model.component.Box;
+import hwr.oop.riddler.model.component.Cell;
 import hwr.oop.riddler.model.component.Column;
 import hwr.oop.riddler.model.component.Row;
 
+import java.util.ArrayList;
+
 public class Sudoku {
+    public Cell[][] cells = new Cell[9][9];
     public Row[] rows = new Row[9];
     public Column[] columns = new Column[9];
-    public Box[] box = new Box[9];
+    public Box[] boxes = new Box[9];
 
     public Sudoku(int[][] input) {
+        this.fillCells(input);
+        this.interpretRows();
+        this.interpretColumns();
+        this.interpretBoxes();
+    }
+
+    private void fillCells(int[][] input) {
+        for (int rowIndex = 0; rowIndex < input.length; rowIndex++) {
+            int[] row = input[rowIndex];
+            for (int columnIndex = 0; columnIndex < row.length; columnIndex++) {
+                int cellValue = row[columnIndex];
+                this.cells[rowIndex][columnIndex] = new Cell(cellValue);
+            }
+        }
+    }
+
+    private void interpretRows() {
+        for (int rowIndex = 0; rowIndex < this.cells.length; rowIndex++) {
+            this.rows[rowIndex] = new Row(cells[rowIndex]);
+        }
+    }
+
+    private void interpretColumns() {
+        for (int columnIndex = 0; columnIndex < this.cells[0].length; columnIndex++) {
+            var column = new ArrayList<Cell>();
+            for (Cell[] row : this.cells) {
+                column.add(row[columnIndex]);
+            }
+
+            var columnArray = new Cell[column.size()];
+            this.columns[columnIndex] = new Column(column.toArray(columnArray));
```

Comment:
Du solltest doch auch einfach nur column.toArray() machen können, oder? Also ohne Zeile 46.

Alternativ einfach nur ein array verwenden und die List ganz sparen:

var columnArray = new Cell[this.cells[0].length];
for (var row = 0; row < this.cells.length; row++) {
    columnArray[this.cells[row][columnIndex]];
}
this.columns[columnIndex] = new Column(columnArray);

Reply:
column.toArray funktioniert leider nicht. Das ist von java leider nicht so smooth geregelt -_-

Reply:
Ahhh okay ja verstehe, dann würde ich aber auf die Liste verzichten oder?